pkg/git/git.go

```diff
@@ -90,6 +93,28 @@ func (h *stiGit) Checkout(repo, ref string) error {
 	return h.runner.RunWithOptions(opts, "git", "checkout", ref)
 }
 
+// GetInfo retrieves the informations about the source code and commit
+func (h *stiGit) GetInfo(repo string) *api.SourceInfo {
+	git := func(arg ...string) string {
+		command := exec.Command("git", arg...)
+		command.Dir = repo
+		out, err := command.CombinedOutput()
+		if err != nil {
+			glog.V(1).Infof("Error executing 'git %#v': %s (%v)", arg, out, err)
+			return ""
+		}
+		return strings.TrimSpace(string(out))
+	}
+	return &api.SourceInfo{
+		Location: git("config", "--get", "remote.origin.url"),
+		CommitID: git("rev-parse", "--abbrev-ref", "HEAD"),
```

Comment:
This returns the ref actually, and the command in ref returns commitID, you need to switch them.